install.sh

```diff
@@ -1,25 +1,98 @@
-#!/bin/sh
-
-echo "Installing latest 'mob' release from GitHub"
-case "$(uname -s)" in
-   Darwin)
-      system="darwin"
-     ;;
-   *)
-      system="linux"
-     ;;
-esac
-url=$(curl -s https://api.github.com/repos/remotemobprogramming/mob/releases/latest \
-| grep "browser_download_url.*mob_.*${system}_amd64\.tar\.gz" \
-| cut -d ":" -f 2,3 \
-| tr -d \")
-# echo "$url"
-tarball="${url##*/}"
-
-curl -sSL $url | tar xz -C /usr/local/bin/ mob && chmod +x /usr/local/bin/mob
-
-location="$(which mob)"
-echo "Mob binary location: $location"
-
-version="$(mob version)"
-echo "Mob binary version: $version"
\ No newline at end of file
+#!/bin/sh   
+target=/usr/local/bin
+local_target=$(systemd-path user-binaries)
+user_arg=$1
+stream_cmd="curl -s https://raw.githubusercontent.com/remotemobprogramming/mob/master/install.sh"
+readme_location="https://github.com/remotemobprogramming/mob/blob/master/README.md"
+
+handle_user_installation() {
+  if [ "$user_arg" = "--user" ]
+  then
+    if [ "$local_target" != "" ] && [ ! -d "$local_target" ]
+    then
+      mkdir $local_target
+    fi
+      
+    if [ -d "$local_target" ]
+    then
+      target=$local_target
+    else
+      echo "unfortunately, there is no user-binaries path on your system. aborting installation."
+      exit 1
+    fi
+  fi
+}
+
+check_access_rights() {
+  if [ ! -w $target ]
+  then
+    echo "you do not have access rights to $target."
+    echo
+    if [ "$local_target" != "" ]
+    then
+      echo "we recommend that you use the --user flag"
+      echo "to install the app into your user binary path $local_target"
+      echo
+      echo "  $stream_cmd | sh -s - --user"
+      echo
+    fi
+    echo "calling the installation with sudo might help."
+    echo "please do it ONLY if you understand the implications of calling some remote script with ROOT rights."
```

Comment:
Would remove this, people know that. With great power comes great responsibility. :-)